sniffles/rulereader.py

```diff
@@ -1,11 +1,10 @@
-import re
-import sys
 import codecs
 import random
+import re
+import sys
+import xml.etree.ElementTree as ET
 from os import listdir
 from os.path import isfile, join
-import xml.etree.ElementTree as ET
-from sniffles.ruletrafficgenerator import *
```

Comment:
getTrafficStreamObject will return a TrafficStream (or derived) object.  TrafficStream is defined in ruletrafficgenerator.py.  How should we best resolve this?  Or is Python smart enough to work it out?

Reply:
First of all, it doesn't cause any problem because `getTrafficStreamObject()` is used in `ruletrafficgenerator.py` only. So, when it is called, `TrafficStream` is always defined.

That is what we should fix in the current module structure: a function in `rulereader.py` uses a class defined in `ruletrafficgenerator.py`, and `ruletrafficgenerator.py` calls a function defined in `rulereader.py`. Any idea to break the cycle?

Reply:
We could remove the polymorphism from rulereader.py and instead implement a dispatch or similar function in the Conversation class rather than making a call to getTrafficStreamObject().  That seems like the simple fix.  Maybe a little hacky though.  Otherwise, I think we need to maybe break conversation out of ruletraffic generator.  Still might need some hackiness to get rid of the cyclic imports.

Reply:
Do we want to do this with this pull request, or make a new pull request to address it?

Reply:
Since this PR doesn't do any harm, I prefer merging this first and then making a new PR.

Reply:
Sounds like a plan.